pkg/subctl/cmd/gather/cni.go

```diff
@@ -107,6 +121,43 @@ func logIPTablesCmds(info Info, pod *v1.Pod) {
 	}
 }
 
+func OVNData(info Info) {
+	if networkPluginCNIType[info.Submariner.Status.NetworkPlugin] != typeOvn {
+		return
+	}
+
+	pods, err := findPods(info.ClientSet, ovnMasterPodLabelOCP)
+	if err != nil && apiErrors.IsNotFound(err) {
```

Comment:
len(pods) is checked in `findponds`. If len = 0 , it returns an error which we are already checking.

Reply:
No it doesn't

```
func findPods(clientSet kubernetes.Interface, byLabelSelector string) (*corev1.PodList, error) {
        pods, err := clientSet.CoreV1().Pods("").List(metav1.ListOptions{LabelSelector: byLabelSelector})

        if err != nil {
                return nil, errors.WithMessage(err, "error listing pods")
        }

        return pods, nil
}

```

Reply:
ohh..that check was removed by a subsequent commit. So yes, the check is needed

Reply:
https://github.com/submariner-io/submariner-operator/commit/a38930deb6c34ae556954ae9183fff3d58a7bcbd

Reply:
List doesn't return IsNotFound? :? Ok  I wanted to make sure we didn't exit because of an error, let me check.

Reply:
Well it can return IsNotFound if the resource kind itself doesn't exist. But we're looking up pods so that won't happen. In this case you want to know if any pods instances were found.